core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -402,10 +409,12 @@ public FolderId getFolderIdByFolderName(ExchangeService exchangeService, String
 
 	@Override
 	public EmailMessage toFile(String filename) throws FileSystemException {
-		ExchangeService exchangeService = getConnection();
+		ExchangeFolderReference reference = asFolderReference(filename);
+		ExchangeService exchangeService = getConnection(reference);
 		boolean invalidateConnectionOnRelease = false;
 		try {
-			ItemId itemId = ItemId.getItemIdFromString(filename);
+			// foldername is filename here
+			ItemId itemId = ItemId.getItemIdFromString(reference.getFolderName());
```

Comment:
Als FolderReference ook voor 'files' gebruikt gaat worden, kan de class beter 'ItemReference' gaan heten

Reply:
Mee eens, alhoewel je binnen de EWS package ook een Item object hebt.

http://officedev.github.io/ews-java-api/docs/releases/api-2.0/apidocs/microsoft/exchange/webservices/data/core/service/item/package-summary.html

Wellicht is 'ExchangeObjectReference' de juiste naam.